### PR TITLE
Watch later

### PR DESCRIPTION
**Pull request by kshitijthatte**
- added wishlist context
- added wishlist routes and controller in backed
- added wishlist services to get, add and remove

### CODE REVIEW

File: src/services/watchlaterServices/addWatchlater.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import axios from "axios";
+
+export const addWatchlater = async (token, video) => {
+  return await axios.post(
+    "/api/user/watchlater",
+    { video },
+    { headers: { authorization: token } }
+  );
+};
```
Comment (BelideNikhil) [inline src/services/watchlaterServices/addWatchlater.js]: you can remove async-await here

**Review by BelideNikhil — APPROVED** [review]
The code looks clean and is easily readable.
The UI is nice.
One suggestion to hide sidebar on login/signup.

File: src/contexts/watchlaterContext.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      (async () => {
+        try {
+          const response = await getWatchlater(token);
+          if (response.status === 200) {
+            const watchlater = response.data.watchlater;
+            setWatchlater(watchlater);
+          }
+        } catch (error) {
+          console.error("ERROR", error);
+          console.log(error.message);
```
Comment (tanveertkd) [inline src/contexts/watchlaterContext.js]: Just a nitpick but you could just use one log statement here.

File: src/pages/SinglePlaylist/HorizontalVideoCard.js [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
 
   const removeVideoHandler = video => {
     switch (playlistTitle) {
       case "Liked Videos":
         return removeLikedVideos(video);
+      case "Watch Later":
+        return removeFromWatchlater(video);
       default:
         return;
     }
```
Comment (tanveertkd) [inline src/pages/SinglePlaylist/HorizontalVideoCard.js:17]: Good implementation, using one handler for both the calls. 👍

**Review by tanveertkd — APPROVED** [review]
This is good to go! Code is well structured and easily readable. 👍

**Pull request merged**